src/Components/Aspire.Azure.Storage.Blobs/AzureBlobStorageContainerSettings.cs

```diff
@@ -28,9 +28,10 @@ void IConnectionStringSettings.ParseConnectionString(string? connectionString)
         const string ContainerName = nameof(ContainerName);
 
         DbConnectionStringBuilder builder = new() { ConnectionString = connectionString };
-        if (builder.TryGetValue(Endpoint, out var endpoint) && builder.TryGetValue(ContainerName, out var containerName))
+        if (builder.TryGetValue(Endpoint, out var endpoint) && endpoint is string endpointValue && builder.TryGetValue(ContainerName, out var containerName))
         {
-            ConnectionString = endpoint.ToString();
+            // endpoint can be a URI (azure deployment) or key/value pairs (emulator)
+            base.ParseConnectionStringInternal(endpointValue);
```

Comment:
I don't think this is right. We shouldn't be stuffing the whole "key/value pairs (emulator)" into the Endpoint property.

Instead for emulator, the ContainerName should just be another property of the connection string, like normal.